src/Makefile.am

```diff
@@ -16,7 +16,9 @@ LOADABLE_MODULES_SOURCES = \
 	LoadableModules.cc \
 	LoadableModules.h
 
-SUBDIRS = mem time debug base anyp helper dns html ftp parser comm error eui acl format clients sbuf servers fs repl store DiskIO proxyp
+# order is important. If anything in a subdir requires a convenience library
+# from another subdir, then it must appear later in this list
+SUBDIRS = mem time debug base anyp helper dns html ftp parser comm error eui sbuf acl format clients servers fs repl store DiskIO proxyp
```

Comment:
To reduce future diffs complexity, please consider splitting into multiple lines, one entry per line. I do not insist on that split.

Reply:
I'll do that in a dedicated future PR